src/Build.UnitTests/BackEnd/CacheSerialization_Tests.cs

```diff
@@ -75,5 +112,32 @@ public void OnlySerializeCacheEntryWithSmallestConfigId(object configCache, obje
                 File.Delete(cacheFile);
             }
         }
+
+        [Theory]
+        [MemberData(nameof(CacheData))]
+        public void OnlySerializeResultsForSpecifiedTargets(object configCache, object resultsCache)
+        {
+            string cacheFile = null;
+            try
+            {
+                cacheFile = FileUtilities.GetTemporaryFile("MSBuildResultsCache");
+                Assert.Null(CacheSerialization.SerializeCaches(
+                    (ConfigCache)configCache,
+                    (ResultsCache)resultsCache,
+                    cacheFile,
+                    ProjectIsolationMode.MessageUponIsolationViolation));
+
+                var result = CacheSerialization.DeserializeCaches(cacheFile);
+                Assert.True(result.ConfigCache.HasConfiguration(1));
+                BuildResult buildResult = result.ResultsCache.GetResultsForConfiguration(1);
+                Assert.True(buildResult.HasResultsForTarget("target1"));
+                Assert.True(buildResult.HasResultsForTarget("target2"));
+                Assert.False(buildResult.HasResultsForTarget("target3"));
```

Comment:
Is this flow correct?

1. line 27 says "this project will build with `target1;target2`
2. line 44 issues a request for `target1;target2;target3`
3. line 62 adds results for `target3`
4. Because of 1, the serialization ignores 3

so we get cached results for `1;2`.

That makes sense but I sure would appreciate some comments describing the setup.

Reply:
Yep that's correct.